### PR TITLE
bump: [#4684] Upgrade Nighwatch

### PR DESCRIPTION
**Pull request by sw-joelmut**
Addresses # 4684
#minor

## Description
This PR updates Nightwatch to its latest version. Modifying the existing Nightwatch files to use the new configuration and browser driver validation.

## Specific Changes
  - Updates Nighwatch from `^2.6.2` to `^3.7.0`.
  - Added nohoist property to workspaces to exclude `@types/selenium-webdriver` as it requires being ES6+.
  - Updated browser functional pipeline to the new structure.
  - Added new Nightwatch config.
  - Added new Nightwatch browser and driver validation.
  - Improved echo page and tests files.
  - 

## Testing
The following image shows the pipeline and the new test structure working as expected.
image
image

### CODE REVIEW

File: testing/browser-functional/nightwatch/utils.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
+const { Builder, Browser } = require('selenium-webdriver');
```
Comment (ceciliaavila) [inline testing/browser-functional/nightwatch/utils.js:1]: Should we include the MS header?

File: testing/browser-functional/nightwatch/requirements.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+const { DEFAULT_BROWSER, getFlag, getBrowser, logs, isBrowserInstalled, isBotRunning } = require('./utils');
```
Comment (ceciliaavila) [inline testing/browser-functional/nightwatch/requirements.js:1]: Should we include the MS header?

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: testing/browser-functional/nightwatch/requirements.js (hunk 2 above)
Comment (sw-joelmut) [reply testing/browser-functional/nightwatch/requirements.js:6]: It didn't have the header before, I added it, thanks!

**Pull request closed without merging**